### ISSUE TITLE
Create Auth ENV config prop

### ISSUE THREAD

**Issue by benjdlambert** [backstage/backstage#3441]
## Feature Suggestion
We want to be able to change the authentication environment for passing onto `AuthProviders` like the Okta Provider https://github.com/backstage/backstage/blob/master/packages/core-api/src/apis/implementations/auth/okta/OktaAuth.ts#L43
## Possible Implementation
Have a top level `auth.environment` key in the config which gets sent to the frontend for use by `AuthProviders`

**Comment by keshann93** [comment]
Interesting project to explore! I am fairly new to the code, but I am sure I can take up this task to get a hang of code base.

**Comment by freben** [comment]
OK, assigned!

**Comment by keshann93** [comment]
Hi @freben Thank you for assigning the issue. I have raised a [PR](https://github.com/backstage/backstage/pull/3523). 

I will try to explain my understanding to give more context on the changes I have made in the PR;

**Intention**
Should be able to override the hard-coded value provided for `environment` attr when calling `static create func` for any auth provider. 

**Solution**
- Added `auth.environment` attr to the top level  of `app-config.yaml` file as mentioned
- Introduced relevant Config schema attributes to pass the value through configApi
- Passed the value through createApiFactory `authProvider1.create` function so default environment value gets overridden with the value passed.
